WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/InPersonPaymentsMenuViewController.swift

```diff
@@ -88,7 +88,7 @@ private extension InPersonPaymentsMenuViewController {
     }
 
     func refreshAfterNewOnboardingState(_ state: CardPresentPaymentOnboardingState) {
-        self.pluginState = nil
+        pluginState = nil
```

Comment:
No, not quite (unless I'm misunderstanding you too!)

I think the references to `self` are leftover from when we moved/copied the onboarding behaviour into this VC, for use with the background onboarding check.

I suspect that they were there because the code was copied/moved from a closure, where `self` capture semantics get enforced by the compiler: you'll have seen the errors where it says you need to explicitly state self.

Now that it's a function, it doesn't capture `self`, so there's no need to put it in.

Reply:
Gotcha! So was just a matter of removing unnecessary `.self` references. The confusion came when I checked the datatype in Xcode for `self.pluginState` vs `pluginState`, and I got back different `InPersonPaymentsMenuViewController` and  `CardPresentPaymentsPluginState`. 

I realize now that the first one is basically `InPersonPaymentsMenuViewController.CardPresentPaymentsPluginState` so effectively the same, and `.self` is redundant.

Reply:
> I realize now that the first one is basically InPersonPaymentsMenuViewController.CardPresentPaymentsPluginState so effectively the same, and .self is redundant.

Sort of...

`InPersonPaymentsMenuViewController.CardPresentPaymentsPluginState` implies that there's a nested type called `CardPresentPaymentsPluginState` inside `InPersonPaymentsMenuViewController`, which there isn't, the view controller simply has a plugin state. I suspect Xcode was just being a bit strange, or giving you the type for `self` rather than `self.pluginState`